api/invite.go

```diff
@@ -526,3 +851,357 @@ func (a *Api) SendInvite(res http.ResponseWriter, req *http.Request, vars map[st
+			ib.Permissions)
+
+		// if the invitee is already a user, we can use his preferences
+		invite.TeamID = ib.TeamID
+		invite.Email = ib.Email
+		invite.IsAdmin = ib.IsAdmin
+		if invitedUsr != nil {
+			invite.UserId = invitedUsr.UserID
+			inviteeLanguage = a.getUserLanguage(invite.UserId, res)
+		}
+
+		var member = store.Member{
+			UserID:           invite.UserId,
+			TeamID:           ib.TeamID,
+			Role:             ib.IsAdmin,
+			InvitationStatus: "pending",
+		}
+		if _, err := a.perms.UpdateTeamMember(tokenValue, member); err != nil {
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusInternalServerError, STATUS_ERR_UPDATING_TEAM)}
+			a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+			return
+		}
+
+		if a.addOrUpdateConfirmation(req.Context(), invite, res) {
+			a.logAudit(req, "invite created")
+
+			if err := a.addProfile(invite); err != nil {
+				log.Println("SendInvite: ", err.Error())
+			} else {
+				var webPath = "hcp/signup"
+
+				// if invitee is already a user (ie already has an account), he won't go to signup but login instead
+				if invite.UserId != "" {
+					webPath = ""
+				}
+
+				emailContent := map[string]interface{}{
+					"MedicalteamName":          team.Name,
+					"MedicalteamAddress":       team.Address,
+					"MedicalteamIentification": team.Code,
+					"CreatorName":              invite.Creator.Profile.FullName,
+					"Email":                    invite.Email,
+					"WebPath":                  webPath,
+				}
+
+				if a.createAndSendNotification(req, invite, emailContent, inviteeLanguage) {
+					a.logAudit(req, "invite sent")
+				} else {
+					a.logAudit(req, "invite failed to be sent")
+					log.Print("Something happened generating an invite email")
+					res.WriteHeader(http.StatusUnprocessableEntity)
+					return
+				}
+			}
+
+			a.sendModelAsResWithStatus(res, invite, http.StatusOK)
+			return
+		}
+	}
+
+}
+
+// @Summary Send notification to an hcp that becomes admin
+// @Description  Send an email to th enew admin
+// @ID hydrophone-api-UpdateTeamInvite
+// @Accept  json
+// @Produce  json
+// @Success 200 {object} models.Confirmation "invite details"
+// @Failure 400 {object} status.Status "userId, teamId and isAdmin were not provided or the payload is missing/malformed"
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provide sufficient privileges"
+// @Failure 403 {object} status.Status "Authorization token is invalid"
+// @Failure 409 {object} status.Status "No notification and email sent; User is already an admin"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
+// @Failure 500 {object} status.Status "Internal error while processing the invite, detailled error returned in the body"// @Router /confirm/send/team-invite [put]
+// @security TidepoolAuth
+func (a *Api) UpdateTeamInvite(res http.ResponseWriter, req *http.Request, vars map[string]string) {
```

Comment:
Can we change the method name. It is confusing, seems like we want to update a invitation record in the confirm table
```suggestion
func (a *Api) ChangeMemberRole(res http.ResponseWriter, req *http.Request, vars map[string]string) {
```